oracle/src/network/proto/mod.rs

```diff
@@ -0,0 +1,156 @@
+use connection::{OracleCommand, OracleConnHandler};
+use reth_eth_wire::{protocol::Protocol, Capability};
+use reth_network::{protocol::ProtocolHandler, Direction};
+use reth_network_api::PeerId;
+use reth_primitives::{Buf, BufMut, BytesMut};
+use std::net::SocketAddr;
+use tokio::sync::mpsc;
+
+pub(crate) mod connection;
+
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) enum OracleProtoMessageId {
+    Ping = 0x00,
+    Pong = 0x01,
+    PingMessage = 0x02,
+    PongMessage = 0x03,
```

Comment:
we now need to figure out what this protocol should actually do @shekhirin

Reply:
A simple (optional: stake weighted) oracle protocol operates in the following phases:
1. Observe data from off-chain source
2. Attest to that data by signing on it
3. Broadcast the signed payload to its peers -- this can happen either on a fixed interval basis, or triggered as a response to an external action e.g. an onchian event being emitted as a result of the "request / callback" pattern, frequently seen in ChainLink or Axiom-style contracts
4. Track signatures
5. Once it hits quorum of signatures, send it out either directly to the place where the data will trigger an action (i.e. it handles also being a "keeper", e.g. submitting the signed oracle 2-of-3 multisig data to a smart contract's callback function, or even archiving it to blobs) or it proceeds to expose it over an (optionally authenticated) API to a user querying it, for self-relaying.

Reply:
Easiest thing to oraclize with this would be Binance API prices.

Reply:
Ok great!

@mattsse can you help me getting this merged? So we keep small iterations?

Reply:
sg!